packages/js/src/ui/components/Notification/DefaultNotification.tsx

```diff
@@ -142,107 +158,10 @@ export const DefaultNotification = (props: DefaultNotificationProps) => {
-              }
-            >
-              <Tooltip.Root>
-                <Tooltip.Trigger
-                  asChild={(childProps) => (
-                    <Button
-                      appearanceKey="notificationUnread__button"
-                      size="iconSm"
-                      variant="ghost"
-                      {...childProps}
-                      onClick={async (e) => {
-                        e.stopPropagation();
-                        await props.notification.unread();
-                      }}
-                    >
-                      <MarkAsUnread class={style('notificationUnread__icon', 'nt-size-3')} />
-                    </Button>
-                  )}
-                />
-                <Tooltip.Content data-localization="notification.actions.unread.tooltip">
-                  {t('notification.actions.unread.tooltip')}
-                </Tooltip.Content>
-              </Tooltip.Root>
-            </Show>
-          </Show>
-          <Show
-            when={props.notification.isArchived}
-            fallback={
-              <Tooltip.Root>
-                <Tooltip.Trigger
-                  asChild={(childProps) => (
-                    <Button
-                      appearanceKey="notificationArchive__button"
-                      size="iconSm"
-                      variant="ghost"
-                      {...childProps}
-                      onClick={async (e) => {
-                        e.stopPropagation();
-                        await props.notification.archive();
-                      }}
-                    >
-                      <MarkAsArchived class={style('notificationArchive__icon', 'nt-size-3')} />
-                    </Button>
-                  )}
-                />
-                <Tooltip.Content data-localization="notification.actions.archive.tooltip">
-                  {t('notification.actions.archive.tooltip')}
-                </Tooltip.Content>
-              </Tooltip.Root>
-            }
-          >
-            <Tooltip.Root>
-              <Tooltip.Trigger
-                asChild={(childProps) => (
-                  <Button
-                    appearanceKey="notificationUnarchive__button"
-                    size="iconSm"
-                    variant="ghost"
-                    {...childProps}
-                    onClick={async (e) => {
-                      e.stopPropagation();
-                      await props.notification.unarchive();
-                    }}
-                  >
-                    <MarkAsUnarchived class={style('notificationArchive__icon', 'nt-size-3')} />
-                  </Button>
-                )}
-              />
-              <Tooltip.Content data-localization="notification.actions.unarchive.tooltip">
-                {t('notification.actions.unarchive.tooltip')}
-              </Tooltip.Content>
-            </Tooltip.Root>
-          </Show>
+          {renderNotificationActions(props.notification, status)}
```

Comment:
I've moved this conditional show/if action button logic to separate file and make it a bit more simpler to understand. The nested <Show> and fallbacks were quite difficult to understand